cloud/aws/step-2/fargate/inputs.tf

```diff
@@ -15,7 +15,19 @@ variable "environment" {
 
 variable "vpc_id" {}
 
-variable "subnet_id" {}
+// 
+variable "subnet_id" {
+  description = "Resources will only accept traffic from within this subnet"
+}
+
+data "aws_subnet" "selected" {
```

Comment:
Yes, I am never 100% sure where to put these. There is probably some sort of convention but I don't know any reference document about it. And if there is, I have no idea how well it is applied out there. My logic here is that datasources qualify as "inputs" as they take data from the outside 😄. As for locals, I also like to keep them there, especially when they are simple transformations of variables. 

I am more than open to define some explicit guidelines, but I think that would be a task in itself.

Reply:
Since we don't have a formal / official way to go about this, we just proceed with the way we've been doing it already.

As for the explicit guidelines, I'll file a ticket, but it is definitely not urgent or high priority.